pkg/cluster/gitrepo/gitrepo_cluster.go

```diff
@@ -1126,6 +1195,306 @@ func (g *clusterGitRepo) UpdateTags(ctx context.Context, application, cluster, t
 	return nil
 }
 
+func (g *clusterGitRepo) DefaultBranch() string {
+	return g.defaultBranch
+}
+
+func (g *clusterGitRepo) UpgradeCluster(ctx context.Context,
```

Comment:
this method likes a script writing in python, please refactor it, take some logic to be a method.